File: socket/python/tcp_echo_server.py
'''
sock.py example
'''

import sock

HOST = ''
PORT = 8322


serveSock = sock.sock('tcp', log=True)
serveSock.bind(HOST, PORT)
serveSock.listen()
s, addr = serveSock.accept()

while True:
    s.send(s.recv_with_delimiter('\n'))
    #s.send(s.recv_fixed_length(3))

s.close()
serveSock.close()

File: socket/java/TcpEchoServer.java
package socket.java;

public class TcpEchoServer {
    static int PORT = 9933;

    public static void main(String[] args) {
        ServSock ss = null;
        TcpSock s = null;
        try {
            ss = new ServSock(PORT, true);
            s = ss.accept();

            while (true)
                s.send(s.recvLine() + "\n");
        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            try {
                if (s != null)
                    s.closeAll();
                if (ss != null)
                    ss.close();
            } catch (Exception e) {}
        }
    }
}
